src/lib.rs

```diff
@@ -369,6 +369,57 @@ impl<T> NonEmpty<T> {
         NonEmpty(f(self.0), self.1.into_iter().map(f).collect())
     }
 
+    /// When we have a function that goes from some `T` to a `NonEmpty<U>`,
+    /// we may want to apply it to a `NonEmpty<T>` but keep the structure flat.
+    /// This is where `and_then` shines.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use nonempty::NonEmpty;
+    ///
+    /// let non_empty = NonEmpty::from((1, vec![2, 3, 4, 5]));
+    ///
+    /// let windows = non_empty.and_then(|i| {
+    ///     let mut next = NonEmpty::new(i + 5);
+    ///     next.push(i + 6);
+    ///     next
+    /// });
+    ///
+    /// let expected = NonEmpty::from((6, vec![7, 7, 8, 8, 9, 9, 10, 10, 11]));
+    ///
+    /// assert_eq!(windows, expected);
+    /// ```
+    pub fn and_then<U, F>(self, mut f: F) -> NonEmpty<U>
```

Comment:
Isn't this basically `flat_map`?

Reply:
Ya, it seems that the naming is inconsistent in Rust. I suppose `flat_map` is more meaningful for the case of `NonEmpty` :)